Update documentation for paths.include/exclude (#2144)
semgrep/semgrep-docs: docs/writing-rules/rule-syntax.md

    paths:
      exclude:
        - "*.jinja2"
        - "**/backend/*_test.go"
        - "/src/tests"
        - /src/static/*.js
```

When invoked with `semgrep -f rule.yaml src` from the Git project
root, the above rule runs
on file paths inside `src/`, but no results are returned for:

- any file with a `.jinja2` file extension;
- any file whose name ends in `_test.go` that exists in a folder named
  `backend` such as the file `src/backend/server_test.go`;
- any file inside `src/tests/` or its subdirectories, or a regular file
  named `src/tests`;
- any file matching the `src/static/*.js` glob pattern such as
  `src/static/hello.js` but not `old/src/static/hello.js`.

The selected set of files to scan with the rule
is independent of the current work
folder. The commands `semgrep -f rule.yaml src` and
`(cd src; semgrep -f rule.yaml .)` will therefore scan the same files.

:::note
The glob syntax conforms to the
[Semgrepignore v2](https://semgrep.dev/docs/semgrepignore-v2-reference)
and Gitignore specifications. Patterns are
matched against the normalized file path relative to the project root as
well as all its parent directories. Beware that the presence of

> a leading slash (e.g. `/*.c`) or a slash in the middle of the pattern

Reviewer: ```suggestion
a leading slash (such as `/*.c`) or a slash in the middle of the pattern
```
For ESL readers

> (e.g. `a/*.c`) anchors the pattern to the project root. `a/*.c` is

Reviewer: ```suggestion
(such as `a/*.c`) anchors the pattern to the project root. `a/*.c` is
```
Yep, technical writing is practically Newspeak/Ingsoc.
